AWSTranscribeStreamingUnitTests/AWSTranscribeStreamingClientWebSocketProviderTests.swift

```diff
@@ -0,0 +1,332 @@
+        
+        let client = AWSTranscribeStreaming(forKey: "testWSSInitWithWebSocketProvider")
+        
+        guard let request = AWSTranscribeStreamingStartStreamTranscriptionRequest() else {
+            XCTFail("request unexpectedly nil")
+            return
+        }
+        
+        request.languageCode = .enUS
+        request.mediaEncoding = .pcm
+        request.mediaSampleRateHertz = 8000
+        
+        let delegate = MockTranscribeStreamingClientDelegate()
+        let receiveConnected = expectation(description: "Callback received connected status")
+        delegate.connectionStatusCallback = { status, error in
+            if status == AWSTranscribeStreamingClientConnectionStatus.connected {
+                receiveConnected.fulfill()
+            }
+        }
+        
+        client.setDelegate(delegate, callbackQueue: DispatchQueue.global())
+        
+        client.startTranscriptionWSS(request)
+        
+        wait(for: [receiveConnected], timeout: 0.1)
+    }
+    
+    // Given: A web socket provider
+    // When: when registering the streaming client
+    // Then: the streaming client is able to use a third party web socket provider and open the socket and disconnect from it
+    func testWSSDisconnect() {
+        let mockWebSocketProvider = MockWebSocketProvider()
+        mockWebSocketProvider.setErrorOnConnectionStatusDidChange(error: nil)
+        guard let config = AWSServiceConfiguration(
+            region: .USEast1,
+            credentialsProvider: MockReturningCredentialsProvider()) else {
+                XCTFail("Couldn't initialize config with MockErrorReturningCredentialsProvider")
+                return
+        }
+        
+        AWSTranscribeStreaming.register(with: config, forKey: "testWSSInitWithWebSocketProvider", webSocketProvider: mockWebSocketProvider)
+        
+        let client = AWSTranscribeStreaming(forKey: "testWSSInitWithWebSocketProvider")
+        
+        guard let request = AWSTranscribeStreamingStartStreamTranscriptionRequest() else {
+            XCTFail("request unexpectedly nil")
+            return
+        }
+        
+        request.languageCode = .enUS
+        request.mediaEncoding = .pcm
+        request.mediaSampleRateHertz = 8000
+        
+        let delegate = MockTranscribeStreamingClientDelegate()
+        
+        let receiveClosed = expectation(description: "Callback received closed status")
+        delegate.connectionStatusCallback = { status, error in
+            if status == AWSTranscribeStreamingClientConnectionStatus.closed {
+                receiveClosed.fulfill()
+            }
+        }
+        
+        client.setDelegate(delegate, callbackQueue: DispatchQueue.global())
+        
+        client.startTranscriptionWSS(request)
+        mockWebSocketProvider.disconnect()
+        wait(for: [receiveClosed], timeout: 10)
```

Comment:
10s seems a bit high for a unit test.

Reply:
oh I dont need that anymore, let me change to something less

Reply:
fixed on all of them